### PR TITLE
Remove ERDS reference affecting `wgs_cnv_annotation` (#257)

### LINKED ISSUE #257: ERDS removal incomplete - affected `wgs_cnv_annotation`
**Issue**
ERDS was removed as an option from WGS CNV calling but not all related workflows were properly adjusted. The workflow  `wgs_cnv_annotation` still expects BCF files instead of VCF.
Error:
```
Missing input files for rule wgs_cnv_annotation_vcf_cnv_filter:
    output: work/bwa.gcnv.annotated.<LIBRARY_NAME>/out/bwa.gcnv.annotated.<LIBRARY_NAME>.vcf.gz, work/bwa.gcnv.annotated.<LIBRARY_NAME>/out/bwa.gcnv.annotated.<LIBRARY_NAME>.vcf.gz.tbi, work/bwa.gcnv.annotated.<LIBRARY_NAME>/out/bwa.gcnv.annotated.<LIBRARY_NAME>.vcf.gz.md5, work/bwa.gcnv.annotated.<LIBRARY_NAME>/out/bwa.gcnv.annotated.<LIBRARY_NAME>.vcf.gz.tbi.md5
    wildcards: mapper=bwa, caller=gcnv, index_ngs_library=<LIBRARY_NAME>
    affected files:
        /path/to/wgs_cnv_calling/work/bwa.gcnv.merge_genotypes/out/bwa.gcnv.merge_genotypes.bcf
        /path/to/wgs_cnv_calling/work/bwa.gcnv.merge_genotypes/out/bwa.gcnv.merge_genotypes.bcf.csi
11-25 17:07 snakemake.logging ERROR    MissingInputException in line 71 of /path/to/snappy-pipeline/snappy_pipeline/workflows/wgs_cnv_annotation/Snakefile:
Missing input files for rule wgs_cnv_annotation_vcf_cnv_filter:
    output: work/bwa.gcnv.annotated.<LIBRARY_NAME>/out/bwa.gcnv.annotated.<LIBRARY_NAME>.vcf.gz, work/bwa.gcnv.annotated.<LIBRARY_NAME>/out/bwa.gcnv.annotated.<LIBRARY_NAME>.vcf.gz.tbi, work/bwa.gcnv.annotated.<LIBRARY_NAME>/out/bwa.gcnv.annotated.<LIBRARY_NAME>.vcf.gz.md5, work/bwa.gcnv.annotated.<LIBRARY_NAME>/out/bwa.gcnv.annotated.<LIBRARY_NAME>.vcf.gz.tbi.md5
    wildcards: mapper=bwa, caller=gcnv, index_ngs_library=<LIBRARY_NAME>
    affected files:
        /path/to/wgs_cnv_calling/work/bwa.gcnv.merge_genotypes/out/bwa.gcnv.merge_genotypes.bcf
        /path/to/wgs_cnv_calling/work/bwa.gcnv.merge_genotypes/out/bwa.gcnv.merge_genotypes.bcf.csi
```

**To Reproduce**
Relevant config parts to reproduce the behavior:
```
  ...
  wgs_cnv_calling:
    path_ngs_mapping: ../ngs_mapping
    tools: [gcnv]
    gcnv:
      precomputed_model_paths:
        - library: "default"
          contig_ploidy: /path/to/bwa.gcnv_contig_ploidy.default/out/bwa.gcnv_contig_ploidy.default/ploidy-model
          model_pattern: /path/to/bwa.gcnv_call_cnvs.default.*/out/bwa.gcnv_call_cnvs.default.*/cnv_calls-model
      path_uniquely_mapable_bed: /path/to/GRCh37/wgEncodeCrgMapabilityAlign75mer.eq_1.bed.gz

  wgs_cnv_annotation:
    tools_ngs_mapping: [bwa]
    tools_wgs_cnv_calling: [gcnv]

  wgs_cnv_export:
    release: GRCh37
    ...
```

**Expected behavior**
The correct file extension and name convention should be set in `wgs_cnv_annotation`.
From `snappy-pipeline/snappy_pipeline/workflows/wgs_cnv_annotation/__init__.py`:
```python
class VcfCnvFilterStepPart(BaseStepPart):

    ...

    def get_input_files(self, action):
        """Return input function returning input file dict."""

        @dictify
        def input_function(wildcards):
            # Validate action
            self._validate_action(action)

            # TODO: make work for non-ERDS/ERDS+SV2
            tpl = "work/write_pedigree.{index_ngs_library}/out/{index_ngs_library}.ped"
            yield "ped", tpl.format(**wildcards)
            tpl = (
                "work/{mapper}.{caller}.{index_ngs_library}/out/"
                "{mapper}.{caller}.{index_ngs_library}"
            )
            if wildcards.caller.startswith("erds"):
                key_ext = {"vcf": ".vcf.gz", "tbi": ".vcf.gz.tbi"}
            else:
                key_ext = {"vcf": ".bcf", "tbi": ".bcf.csi"}
            # SVs
            wgs_cnv_calling = self.parent.sub_workflows["wgs_cnv_calling"]
            if wildcards.caller == "erds_sv2":
                tpl = tpl.replace(".{index_ngs_library}", ".merge_genotypes")
            elif not wildcards.caller.startswith("erds"):
                tpl = tpl.replace(".{index_ngs_library}", ".merge_genotypes")
            for key, ext in key_ext.items():
                yield key, wgs_cnv_calling(tpl + ext).format(**wildcards)
            return

        return 
```

**Additional context**
Removed ERDS: #215 
Added gCNV: #190 
Added Delly2: #186

### PR DESCRIPTION
**Pull request by eudesbarbosa**
Closes #257

### CODE REVIEW

**Comment by Nicolai-vKuegelgen** [comment]
Looking good to me

File: tests/snappy_pipeline/workflows/test_workflows_wgs_cnv_annotation.py [hunk 1]
```diff
@@ -95,11 +94,11 @@ def wgs_cnv_annotation_workflow(
 def test_vcf_cnv_filter_step_part_get_input_files(wgs_cnv_annotation_workflow):
     """Tests VcfCnvFilterStepPart.get_input_files()"""
     wildcards = Wildcards(
-        fromdict={"mapper": "bwa", "index_ngs_library": "P001-N1-DNA1-WGS1", "caller": "erds_sv2"}
+        fromdict={"mapper": "bwa", "index_ngs_library": "P001-N1-DNA1-WGS1", "caller": "delly2"}
     )
     # Define expected
     wgs_cnv_calling_path = (
-        "WGS_CNV_CALLING/work/bwa.erds_sv2.merge_genotypes/out/bwa.erds_sv2.merge_genotypes"
+        "WGS_CNV_CALLING/output/bwa.delly2.P001-N1-DNA1-WGS1/out/bwa.delly2.P001-N1-DNA1-WGS1"
```
Comment (Nicolai-vKuegelgen) [inline tests/snappy_pipeline/workflows/test_workflows_wgs_cnv_annotation.py:101]: Why is only delly2 and not also gcnv added here? Both are defined in the minimal_config.

File: tests/snappy_pipeline/workflows/test_workflows_wgs_cnv_annotation.py [hunk 2]
```diff
@@ -95,11 +94,11 @@ def wgs_cnv_annotation_workflow(
 def test_vcf_cnv_filter_step_part_get_input_files(wgs_cnv_annotation_workflow):
     """Tests VcfCnvFilterStepPart.get_input_files()"""
     wildcards = Wildcards(
-        fromdict={"mapper": "bwa", "index_ngs_library": "P001-N1-DNA1-WGS1", "caller": "erds_sv2"}
+        fromdict={"mapper": "bwa", "index_ngs_library": "P001-N1-DNA1-WGS1", "caller": "delly2"}
```
Comment (Nicolai-vKuegelgen) [inline tests/snappy_pipeline/workflows/test_workflows_wgs_cnv_annotation.py:97]: Why is only delly2 and not also gcnv added here? Both are defined in the minimal_config.

File: snappy_pipeline/workflows/wgs_cnv_annotation/__init__.py [hunk 3]
```diff
@@ -114,27 +114,18 @@ def get_input_files(self, action):
...
-
-            # TODO: make work for non-ERDS/ERDS+SV2
+            # Pedigree file
             tpl = "work/write_pedigree.{index_ngs_library}/out/{index_ngs_library}.ped"
             yield "ped", tpl.format(**wildcards)
+            # SVs
+            key_ext = {"vcf": ".vcf.gz", "tbi": ".vcf.gz.tbi"}
             tpl = (
-                "work/{mapper}.{caller}.{index_ngs_library}/out/"
+                "output/{mapper}.{caller}.{index_ngs_library}/out/"
```
[the remark this answers is not in the document]
Comment (eudesbarbosa) [reply snappy_pipeline/workflows/wgs_cnv_annotation/__init__.py:123]: Unwritten convention - `output` just has symlinks, but it has the files that should be somehow exposed to other workflows or transfer to SODAR. In short, the "final product" of a workflow.

[the remark this answers is not in the document]
Comment (holtgrewe) [reply snappy_pipeline/workflows/wgs_cnv_annotation/__init__.py:123]: Actually, it's a [written rule](https://github.com/bihealth/snappy-pipeline/blob/4969c4a5f31b100ac74356ad44a31de4e166f10c/docs/step_generic.rst#directory-output) ;-)

File: tests/snappy_pipeline/workflows/test_workflows_wgs_cnv_annotation.py (hunk 2 above)
Comment (eudesbarbosa) [reply tests/snappy_pipeline/workflows/test_workflows_wgs_cnv_annotation.py:97]: This tests a single call from a Snakemake rule, so single values for each wildcard. I could have created a different test for `gcnv`, but it seemed repetitive.

File: tests/snappy_pipeline/workflows/test_workflows_wgs_cnv_annotation.py (hunk 1 above)
Comment (eudesbarbosa) [reply tests/snappy_pipeline/workflows/test_workflows_wgs_cnv_annotation.py:101]: Only `delly2` defined in the test wildcard object - single test.

**Review by Nicolai-vKuegelgen — APPROVED** [review]

**Pull request merged**